Framework/DataHandling/src/LoadMuonNexusV2.cpp

```
@@ -0,0 +1,227 @@
+                  "0 indicates that every entry is loaded, into a separate "
+                  "workspace within a group. "
+                  "A positive number identifies one entry to be loaded, into "
+                  "one workspace");
+
+  std::vector<std::string> FieldOptions{"Transverse", "Longitudinal"};
+  declareProperty("MainFieldDirection", "Transverse",
+                  std::make_shared<StringListValidator>(FieldOptions),
+                  "Output the main field direction if specified in Nexus file "
+                  "(default longitudinal).",
+                  Direction::Output);
+
+  declareProperty("TimeZero", 0.0,
+                  "Time zero in units of micro-seconds (default to 0.0)",
+                  Direction::Output);
+  declareProperty("FirstGoodData", 0.0,
+                  "First good data in units of micro-seconds (default to 0.0)",
+                  Direction::Output);
+
+  declareProperty(
+      std::make_unique<WorkspaceProperty<Workspace>>(
+          "DeadTimeTable", "", Direction::Output, PropertyMode::Optional),
+      "Table or a group of tables containing detector dead times.");
+
+  declareProperty(std::make_unique<WorkspaceProperty<Workspace>>(
+                      "DetectorGroupingTable", "", Direction::Output,
+                      PropertyMode::Optional),
+                  "Table or a group of tables with information about the "
+                  "detector grouping.");
+}
+void LoadMuonNexusV2::execLoader() {
+  this->setRethrows(true);
+  // prepare nexus entry
+  m_entrynumber = getProperty("EntryNumber");
+  m_filename = getPropertyValue("Filename");
+  NXRoot root(m_filename);
+  NXEntry entry = root.openEntry("raw_data_1");
+
+  // Execute child algorithm LoadISISNexus2 and load Muon specific properties
+  runLoadISISNexus();
+  loadMuonProperties(entry);
+
+  // Check if single or multi period file and create appropriate loading
+  // strategy
+  isEntryMultiPeriod(entry);
```

Should this check be first and throw an exception instead if a multi-period file is encountered? Failing early and loud is usually better if we can't do something.